src/shared_utilities/SpaceVaryingInputs.jl

```diff
@@ -0,0 +1,133 @@
+# SpaceVaryingInputs.jl
+#
+# This module contains methods to process external data, regrid it onto the
+# model grid, and return the corresponding fields for use in the simulation.
+# This module only concerns with external data which varies in space,
+# and not time. For temporally varying input, we refer you to `TimeVaryingInputs.jl`.
+
+# All spatially varying parameter fields are assumed to fit into memory.
+
+# The planned parameter underlying arrays are:
+# - one-dimensional (values prescribed as a function of depth at a site),
+# - two-dimensional (values prescribed globally at each lat/lon),
+# - three-dimensional (values prescribed as a function of depth globally)
+# - analytic (functions of the coordinates of the space)
+
+module SpaceVaryingInputs
+using ClimaCore
+using DocStringExtensions
+import ..searchsortednearest
+import ..linear_interpolation
+using ClimaLand.FileReader
+export SpaceVaryingInput
+
+# Analytic case
+"""
+    SpaceVaryingInput(data_function::Function, space::S) 
+    where {S <: ClimaCore.Spaces.AbstractSpace}
+  
+Returns the parameter field to be used in the model; appropriate when
+a parameter is defined using a function of the coordinates of the space.
+
+Pass the ``data" as a function `data_function` which takes coordinates as arguments, 
+and  the ClimaCore space of the model simulation.
+"""
+function SpaceVaryingInput(
+    data_function::Function,
+    space::ClimaCore.Spaces.AbstractSpace,
+)
+    model_value = ClimaCore.Fields.zeros(space)
+    coords = ClimaCore.Fields.coordinate_field(space)
+    @. model_value = data_function(coords)
```

Comment:
before, we wrapped the function call in `FT`. Do we still need to do that here?

Reply:
Good question. I removed it here, but then in the tests I had to add it back. For example, if we have f = (coords) -> 2, we cant broadcast that over coordinates which correspond to a floating point field:

```
julia> column = Domains.Column(; zlim = zlim, nelements = nelements[3])
ClimaLand.Domains.Column{Float32}((-1.0f0, 0.0f0), (10,), nothing, (:bottom, :top), (surface = ClimaCore.Spaces.PointSpace{ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}, ClimaCore.DataLayouts.DataF{ClimaCore.Geometry.LocalGeometry{(3,), ClimaCore.Geometry.ZPoint{Float32}, Float32, StaticArraysCore.SMatrix{1, 1, Float32, 1}}, SubArray{Float32, 1, Matrix{Float32}, Tuple{Int64, Base.Slice{Base.OneTo{Int64}}}, true}}}(ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded}(ClimaComms.CPUSingleThreaded()), ClimaCore.DataLayouts.DataF{ClimaCore.Geometry.LocalGeometry{(3,), ClimaCore.Geometry.ZPoint{Float32}, Float32, StaticArraysCore.SMatrix{1, 1, Float32, 1}}, SubArray{Float32, 1, Matrix{Float32}, Tuple{Int64, Base.Slice{Base.OneTo{Int64}}}, true}}
  Float32[0.0, 0.1, 0.05, 10.0, 0.1, 10.0, 100.0, 0.01]), subsurface = CenterFiniteDifferenceSpace:
  context: SingletonCommsContext using CPUSingleThreaded
  mesh: 10-element IntervalMesh of IntervalDomain: z ∈ [-1.0,0.0] (:bottom, :top)))

julia> f = (coords) -> 2
#1 (generic function with 1 method)

julia> space = column.space.subsurface
CenterFiniteDifferenceSpace:
  context: SingletonCommsContext using CPUSingleThreaded
  mesh: 10-element IntervalMesh of IntervalDomain: z ∈ [-1.0,0.0] (:bottom, :top)

julia> coords = Fields.coordinate_field(space)
ClimaCore.Geometry.ZPoint{Float32}-valued Field:
  z: Float32[-0.95, -0.85, -0.75, -0.65, -0.55, -0.45, -0.35, -0.25, -0.15, -0.05]

julia> f.(coords)
ERROR: Struct type Int64 cannot be represented using base type Float32
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] macro expansion
   @ ~/.julia/packages/ClimaCore/wYSFf/src/DataLayouts/struct.jl:34 [inlined]
 [3] check_basetype(::Type{Float32}, ::Type{Int64})
   @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/wYSFf/src/DataLayouts/struct.jl:33
 [4] (ClimaCore.DataLayouts.VF{Int64})(array::Matrix{Float32})
   @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/wYSFf/src/DataLayouts/DataLayouts.jl:900
 [5] similar(bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.VFStyle{Array{…}}, Nothing, var"#1#2", Tuple{ClimaCore.DataLayouts.VF{…}}}, ::Type{Int64})
   @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/wYSFf/src/DataLayouts/broadcast.jl:279
 [6] similar(bc::Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{…}, ClimaCore.Spaces.FiniteDifferenceSpace{…}, var"#1#2", Tuple{…}}, ::Type{Int64})
   @ ClimaCore.Fields ~/.julia/packages/ClimaCore/wYSFf/src/Fields/broadcast.jl:153
 [7] copy(bc::Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{…}, ClimaCore.Spaces.FiniteDifferenceSpace{…}, var"#1#2", Tuple{…}})
   @ ClimaCore.Fields ~/.julia/packages/ClimaCore/wYSFf/src/Fields/broadcast.jl:106
 [8] materialize(bc::Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{ClimaCore.DataLayouts.VFStyle{…}}, Nothing, var"#1#2", Tuple{ClimaCore.Fields.Field{…}}})
   @ Base.Broadcast ./broadcast.jl:903
 [9] top-level scope
   @ REPL[21]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

So, I will add the FT back into src

Reply:
this is interesting! but if I do `model_value .= f.(coords)`, the broadcasting also does the type conversion. @Sbozzolo so it seems like, at least for the analytic input, we need to do 
```
    model_value = ClimaCore.Fields.zeros(space)
    coords = ClimaCore.Fields.coordinate_field(space)
    return model_value .= data_function.(coords)
```

then we can remove the FT from src

Reply:
Please, add comments explaining why you do that :)